### ISSUE TITLE
System Config: Set proper startup scripts and order of startup for test/dev vms.

### ISSUE THREAD

**Issue by kcondon** [IQSS/dataverse#1406]
Recently an ops patch and restart showed us our service start up procedure isn't fully functioning: some services were not started and the order is not specified. Additionally, need to handle secure linux.

These are the steps, roughly speaking, that need to occur for proper startup:
1. service postgresql restart (wait for it to complete)
2. setenforce permissve
3. service httpd restart
4. service shibd restart
5. service glassfish stop (if running), start
6. cd /usr/local/solr/solr-4.6.0/example/
    java -jar -server start.jar &

Most important are demo and build. If the process is documented I can update all other vms.

**Comment by pdurbin** [comment]
Sadly, because we are using the `mod_shib` Apache module we **can't** run with SELinux in "Enforcing" mode:

> "At the present time, we do not support the SP in conjunction with SELinux, and at minimum we know that communication between the mod_shib and shibd components will fail if it's enabled. Other problems may also occur." -- https://wiki.shibboleth.net/confluence/display/SHIB2/NativeSPSELinux

**Comment by landreev** [comment]
I'll attend to this when I have time. 
But this is not a beta 12 issue.

**Comment by pdurbin** [comment]
> If the process is documented I can update all other vms

@kcondon please let me know if the process is documented well enough in my pull request at #2895

Here's is where the especially relevant sections can be previewed:

- http://guides.dataverse.org/en/2884-install-guide-reorg/installation/prerequisites.html#glassfish-init-script
- http://guides.dataverse.org/en/2884-install-guide-reorg/installation/prerequisites.html#postgresql-init-script
- http://guides.dataverse.org/en/2884-install-guide-reorg/installation/prerequisites.html#solr-init-script
- http://guides.dataverse.org/en/2884-install-guide-reorg/installation/shibboleth.html#disable-selinux

I'm realizing that the Shibboleth page doesn't really mention `chkconfig httpd on` to make Apache starts at boot so if you want to kick this back to me to add that I certainly can.

**Comment by kcondon** [comment]
@pdurbin 
Thank Phil this documents the individual components (maybe add that bit about chkconfig for Apache). 
My original intent for this ticket was mostly for internal use though it is also useful for others. 

The main objective that I think still needs to be addressed somewhere is how to set up your server so on rebooting the box, dataverse as an app comes up (all services restart in the correct order).

I did not see that part mentioned specifically. I mentioned the order above, is it just a matter of mentioning the order somewhere or referring to the chkconfig process order?

**Comment by pdurbin** [comment]
@kcondon the order of startup (at least as of RHEL/CentOS 6) can been seen by executing this command:

`ls /etc/rc.d/rc3.d | grep ^S | egrep 'postgres|shibd|httpd|glassfish|solr'`

What that shows on the demo server is this:

```
S64postgresql
S80shibd
S85httpd
S99glassfish
S99solr
```

This means that PostgreSQL is started before Shibboleth (shibd), which is started before Apache (httpd), etc. Glassfish and Solr are started last.

These values come from init scripts, such as "chkconfig: 2345 99 01" in the one linked from http://guides.dataverse.org/en/2884-install-guide-reorg/installation/prerequisites.html#glassfish-init-script

So basically, by providing init scripts for Glassfish and Solr and suggesting the standard startup order for init scripts provided by the operating system (postgres, etc.) we are already specifying the startup order. We can, of course, document this somewhere if you want. Please just pass this back to me if you'd like me to take a swing at this.

**Comment by kcondon** [comment]
Thanks, the explanation is enough if the result is what we want. Closing.
